packages/components/forms/src/text-input/TextInput.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useState, ChangeEvent } from 'react';
+import { BaseInput } from '@contentful/f36-inputs';
+import { Box, Flex } from '@contentful/f36-core';
+import { Label } from '@contentful/f36-forms';
+import { ValidationMessage } from '@contentful/f36-validation-message';
+import { HelpText } from '@contentful/f36-helptext';
+import { CopyButton } from '@contentful/f36-copybutton';
+import getStyles from './TextInput.styles';
+import { TextInputProps } from './types';
+
+export const _TextInput = (
+  {
+    className,
+    testId = 'cf-ui-text-field',
+    countCharacters = false,
+    isStandalone = false,
+    isRequired = false,
+    helpText,
+    label,
+    id,
+    link,
+    validationMessage,
+    value,
+    onChange,
+    placeholder,
+    maxLength,
+    isInvalid,
+    isDisabled,
+    isReadOnly,
+    withCopyButton,
+    onCopy,
+    name,
+    icon,
+    ...otherProps
+  }: TextInputProps,
+  ref: React.Ref<HTMLInputElement>,
+) => {
+  const [valueState, setValueState] = useState<string | undefined>(value);
+  const styles = getStyles({ isInvalid, isDisabled });
+
+  // Store a copy of the value in state.
+  // This is used by this component when the `countCharacters`
+  // option is on
+  const handleOnChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setValueState(e.target.value);
+      if (onChange) onChange(e);
+    },
+    [onChange],
+  );
```

Comment:
I think you don't need this `useCallback` here.... I might be wrong

I guess if the person using this component wants to optimise it, they can pass a `useCallback` as the value of their `onChange` props

But I'm basing my opinion on [this article](https://kentcdodds.com/blog/usememo-and-usecallback) which I may not have understood correctly hahahaha

Reply:
But if we do just 
```
const handleOnChange =     (e: ChangeEvent<HTMLInputElement>) => {
      setValueState(e.target.value);
      if (onChange) onChange(e);
    }
```
(Or you had some different approach in mind?)
then we gonna get a warning `A component is changing an uncontrolled input of type text to be controlled`

Reply:
Since this handleOnChange is passed down to the `BaseInput`, isn't it better to have it inside the useCallback?? so the reference is always the same instead of creating a new anonymous function every render?? So React would use the same reference to the function??

Reply:
@massao cool! thanks for the feedback! okidoks, `useCallback` it is then.